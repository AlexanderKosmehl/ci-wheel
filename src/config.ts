export const WHEEL_GAP_IN_PX = 5;
export const SPIN_BUTTON_SIZE_IN_REM = 3;
export const SPINNER_TICK_OFFSET_IN_REM = 1.5;

export const MIN_SPINS_IN_DEG = 720;
export const SPIN_DURATION_IN_SEC = 2;
export const SEGMENT_COLORS = ['#00BEFF', '#FF1314', '#FFE801', '#72C801'];
