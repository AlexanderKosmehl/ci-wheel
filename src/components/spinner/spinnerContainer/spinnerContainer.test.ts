import {
  describe, expect, it, vi,
} from 'vitest';
import { SPIN_DURATION_IN_SEC } from '../../../config';
import generateSpinnerComponent from './spinnerContainer';

describe('generateSpinnerComponent', () => {
  const spinCallback = vi.fn((_result: string) => {});
  const labels = ['Test', 'Test2'];

  const spinnerComponent = generateSpinnerComponent({
    labels,
    spinCallback,
  });

  // Callback is called after a delay
  vi.useFakeTimers();

  it('generates component correctly', () => {
    expect(spinnerComponent).toMatchSnapshot();
  });

  it('adds working callback', () => {
    const button = spinnerComponent.querySelector<HTMLButtonElement>('[data-test=spinnerButton]');
    if (!button) throw Error('No button rendered!');

    button.click();

    vi.advanceTimersByTime(SPIN_DURATION_IN_SEC * 1000);

    expect(spinCallback).toBeCalled();
    expect(labels).toContain(spinCallback.mock.calls[0][0]);
  });
});
