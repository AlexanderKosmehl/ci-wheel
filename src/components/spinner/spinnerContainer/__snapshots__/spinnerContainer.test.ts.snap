// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSpinnerComponent generates component correctly 1`] = `
<div
  class="container"
>
  <div
    class="spinner"
    style="transition: 2s;"
  >
    <div
      class="segment"
      style="transform: rotate(0deg); clip-path: polygon(
      calc(50% + 5px) 0%,
      100% 0%,
      100% 100%,
      calc(50% + 5px) 100%
    ); background-color: rgb(0, 190, 255);"
    >
      <span
        class="label"
        style="width: calc(50% - 1.5rem - 48px);"
      >
        Test
      </span>
    </div>
    <div
      class="segment"
      style="transform: rotate(180deg); clip-path: polygon(
      calc(50% + 5px) 0%,
      100% 0%,
      100% 100%,
      calc(50% + 5px) 100%
    ); background-color: rgb(255, 19, 20);"
    >
      <span
        class="label"
        style="width: calc(50% - 1.5rem - 48px);"
      >
        Test2
      </span>
    </div>
  </div>
  <div
    class="tickShadow"
  >
    <div
      class="tick"
    />
  </div>
  <div
    class="wrapper"
  >
    <button
      class="button"
    >
      Spin
    </button>
  </div>
</div>
`;
